pkg/px-resource-gateway/priority_queue.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	pb "github.com/libopenstorage/operator/proto"
+)
+
+var (
+	priorityDescendingOrder = []pb.SemaphoreAccessPriority_Type{
+		pb.SemaphoreAccessPriority_HIGH,
+		pb.SemaphoreAccessPriority_MEDIUM,
+		pb.SemaphoreAccessPriority_LOW,
+	}
+)
+
+type PriorityQueue interface {
+	Enqueue(clientId string, priority pb.SemaphoreAccessPriority_Type)
+	Dequeue(priority pb.SemaphoreAccessPriority_Type)
+	Front() (string, pb.SemaphoreAccessPriority_Type)
+	Remove(clientId string)
+}
+
+type priorityQueue struct {
+	sync.Mutex
```

Comment:
interesting style also. Never seen it. Normally it is a variable in the struct and not an embedding. Ok, by me, just haven't seen it.